avocado/core/loader.py

```diff
@@ -44,6 +44,52 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags, ignore_empty=False):
+    """
+    Filter the existing (unfiltered) test suite based on tags
+
+    The filtering mechanism is limited to INSTRUMENTED tests, that is it does
+    not apply to SIMPLE tests or any other test type.  It mean non-INSTRUMENTED
+    tests are *always* included.
+
+    :param test_suite: the unfiltered test suite
+    :type test_suite: dict
+    :param filter_by_tags: the list of tag sets to use as filters
+    :type filter_by_tags: list of comma separated tags (['foo,bar', 'fast'])
+    :param ignore_empty: if instrumented tests without tags should not be
+                         filtered out
+    :param ignore_empty: bool
+    """
+    filtered = []
+    for klass, info in test_suite:
+        if not isinstance(klass, str):  # not instrumented: no filtering
+            filtered.append((klass, info))
+            continue
+
+        test_tags = info['tags']
+        if not test_tags and ignore_empty:       # not tagged: no filtering
```

Comment:
Actually if possible I'd prefer the use of `None` in tags discovery, because there is a great difference between empty and no tags. IMO the tests with `tags=` should be filtered even when `ignore_empty` is set.